src/observable/ConnectableObservable.ts

```diff
@@ -17,28 +17,35 @@ export class ConnectableObservable<T> extends Observable<T> {
     return this._getSubject().subscribe(subscriber);
   }
 
-  _getSubject() {
-    const subject = this.subject;
+  protected _getSubject() {
+    const subject = this._subject;
     if (subject && !subject.isUnsubscribed) {
       return subject;
     }
-    return (this.subject = this.subjectFactory());
+    return (this._subject = this.subjectFactory());
   }
 
   connect(): Subscription {
     const source = this.source;
-    let subscription = this.subscription;
+    let subscription = this._subscription;
     if (subscription && !subscription.isUnsubscribed) {
       return subscription;
     }
     subscription = source.subscribe(this._getSubject());
     subscription.add(new ConnectableSubscription(this));
-    return (this.subscription = subscription);
+    return (this._subscription = subscription);
   }
 
   refCount(): Observable<T> {
     return new RefCountObservable(this);
   }
+
+  // This method is opened for `ConnectableSubscription`.
+  // Not to call from others.
+  _closeSubscription(): void {
```

Comment:
Though it's specific usage only from certain caller it's public interface, would like to avoid `_`. Hope to have something similar to `internal` modifier in TypeScript in further.